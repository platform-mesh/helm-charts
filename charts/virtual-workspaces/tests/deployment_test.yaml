suite: operator
chart:
  appVersion: 0.0.0
release:
  name: virtual-workspaces
tests:
  - it: virtual-workspaces match the snapshot
    set:
      cert:
        extraDnsNames:
          - my.example.com
          - my.example2.com
    asserts:
      - matchSnapshot: {}
  