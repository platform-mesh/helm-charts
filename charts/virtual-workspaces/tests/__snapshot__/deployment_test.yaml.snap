virtual-workspaces match the snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: virtual-workspaces
    spec:
      replicas: null
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: virtual-workspaces
      template:
        metadata:
          annotations: null
          labels:
            app: virtual-workspaces
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - start
                - --tls-cert-file=/certs/tls.crt
                - --tls-private-key-file=/certs/tls.key
                - --kubeconfig=/api-kubeconfig/kubeconfig
                - --bind-address=0.0.0.0
                - --authentication-kubeconfig=/api-authentication-kubeconfig/kubeconfig
                - --authentication-skip-lookup
                - --client-ca-file=/client-ca/tls.crt
                - --requestheader-client-ca-file=/requestheader-client-ca/tls.crt
                - --requestheader-username-headers=X-Remote-User
                - --requestheader-group-headers=X-Remote-Group
                - --requestheader-extra-headers-prefix=X-Remote-Extra-
                - --secure-port=8443
                - --server-url=https://kcp-front-proxy.openmfp-system:8443
              image: ghcr.io/platform-mesh/virtual-workspaces:0.0.0
              imagePullPolicy: IfNotPresent
              name: virtual-workspaces
              ports:
                - containerPort: 8443
                  name: vws
                  protocol: TCP
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /api-kubeconfig
                  name: external-api-server
                - mountPath: /api-authentication-kubeconfig
                  name: authentication
                - mountPath: /client-ca
                  name: client-ca-secret
                - mountPath: /certs
                  name: vws-cert
                - mountPath: /requestheader-client-ca
                  name: requestheader-client-ca
          imagePullSecrets:
            - name: github
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: virtual-workspaces
          terminationGracePeriodSeconds: 10
          volumes:
            - name: external-api-server
              secret:
                secretName: account-operator-kubeconfig
            - name: authentication
              secret:
                secretName: portal-kubeconfig
            - name: client-ca-secret
              secret:
                secretName: kcp-ca
            - name: vws-cert
              secret:
                secretName: kcp-virtual-workspaces-cert
            - name: requestheader-client-ca
              secret:
                secretName: kcp-requestheader-client-ca
  2: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      name: virtual-workspaces
      namespace: NAMESPACE
    spec:
      maxReplicas: 10
      metrics:
        - resource:
            name: cpu
            target:
              averageUtilization: 80
              type: Utilization
          type: Resource
      minReplicas: 1
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: virtual-workspaces
  3: |
    apiVersion: v1
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      name: virtual-workspaces
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      name: virtual-workspaces
    spec:
      ports:
        - port: 8443
          protocol: TCP
          targetPort: 8443
      selector:
        app: virtual-workspaces
